articles/how-to-build-a-todo-app/index.md

```diff
@@ -0,0 +1,216 @@
+
+### Flutter Widgets
+Flutter Widgets describe what the view of an app should look like given the current configuration and state. According to the Flutter documentation, Flutter widgets are built using a modern framework that takes inspiration from React.
+
+A widget can help with layout, define design, etc. For example: Padding, Margin, Center, Layout rows, and columns are all widgets.
+
+From our code, the whole app is a widget which contains a MaterialApp widget.
+- The scaffold is the widget that helps us create a proper material layout without the worry of manual styling.
+- The AppBar is a widget that accepts a title and creates a bar at the top of the screen, this is normal in apps. This aligns the text to the left on Android and aligns text to the center on iOS. 
+
+### Stateless Widget
+A stateless widget is a widget whose state cannot be changed once it is built. i.e. no amount of change in the variables, icons, buttons, or retrieving data can change the state of the app.
+A to-do app will always have to-do items added and remove, to achieve this we will need to implement a stateful widget.
+
+### Stateful Widget
+This type of widget is dynamic. This means it can change its appearance when it receives data or change appearance in response to events triggered.
+
+Edit your code to match the one in the picture below.
+
+![Stateful Widget](/engineering-education/how-to-build-a-todo-app/stateful.jpg)
+  
+You should notice that our stateful widgets class TodoList has two classes. This is to enable us to update our to-do list without losing our data. Now, let us add functionality to the state class. Add the below code to your state:
+
+![Code Picture](/engineering-education/how-to-build-a-todo-app/listText.jpg)
+
+This code final List<String> _todoList = <String>[];. Enable us save data into our app. While 
+final TextEditingController _textFieldController = TextEditingController();
+Is enabling us have a text field for items input. 
+Next, update your code to the one below
+
+![addTodoItem Function](/engineering-education/how-to-build-a-todo-app/function.jpg)
+
+The _addtodoItems function is responsible for saving items into _todolist.
+Update your code again to match the one below.
+
+![ListTile Widget](/engineering-education/how-to-build-a-todo-app/listTile.jpg)
+
+the Widget ListTile is usually to populate a listView in flutter.
+To type in todo items update the code to the one below.
+
+![DisplayDialog](/engineering-education/how-to-build-a-todo-app/displayDialog.jpg)
+
+The alert dialog tells the user about situations that require confirmation. We are going to use the alert box to collect todo Items.
+To run the code we would have to update the build widget code to the one below.
+
+
+![Build Image](/engineering-education/how-to-build-a-todo-app/build.jpg)
+
+Your complete code should look the one below:
+
+```dart
```

Comment:
Please add code comments throughout your code and explain in detail how they work.

Reply:
@cyberShaw I have effected the changes.